tests/src/python/test_provider_ogr_general.py

```diff
@@ -0,0 +1,980 @@
+            vl_view_point = QgsVectorLayer(u'{}|layerid=1|layername=view_test_geom1|featurescounted=1|geometrytype=Point|ogrtype=0'.format(datasource), u'view_point', u'ogr')
+            vl_view_linestring = QgsVectorLayer(u'{}|layerid=2|layername=view_test_geom2|featurescounted=1|geometrytype=LineString|ogrtype=0'.format(datasource), u'view_linestring', u'ogr')
+            features_01 = [f_iter for f_iter in vl_test_point.getFeatures(QgsFeatureRequest().setFilterExpression("ogc_fid = 1"))]
+            test_point_geom = [f_iter.geometry() for f_iter in features_01][0].geometry()
+            features_02 = [f_iter for f_iter in vl_test_linestring.getFeatures(QgsFeatureRequest().setFilterExpression("ogc_fid = 1"))]
+            test_linestring_geom = [f_iter.geometry() for f_iter in features_02][0].geometry()
+            features_03 = [f_iter for f_iter in vl_view_point.getFeatures(QgsFeatureRequest().setFilterExpression("pk_id = 1"))]
+            view_point_geom = [f_iter.geometry() for f_iter in features_03][0].geometry()
+            features_04 = [f_iter for f_iter in vl_view_linestring.getFeatures(QgsFeatureRequest().setFilterExpression("pk_id = 1"))]
+            view_linestring_geom = [f_iter.geometry() for f_iter in features_04][0].geometry()
+            print('-I-> Testing if the Point of the SpatialTable is the same as in the SpatialView. [{0}]'.format('SRID=4326;POINT(0 1)'))
+            self.assertEquals((test_point_geom.x(), test_point_geom.y()), (view_point_geom.x(), view_point_geom.y()))
+            print('-I-> Testing if the LineString of the SpatialTable is the same as in the SpatialView. [{0}]'.format('SRID=4326;LINESTRING(0 1,2 3)'))
+            self.assertEqual((test_linestring_geom.pointN(0).x(), test_linestring_geom.pointN(0).y(),test_linestring_geom.pointN(1).x(), test_linestring_geom.pointN(1).y()),
+                                        (view_linestring_geom.pointN(0).x(), view_linestring_geom.pointN(0).y(),view_linestring_geom.pointN(1).x(), view_linestring_geom.pointN(1).y()))
+            del vl_test_point
+            del vl_test_linestring
+            del vl_view_point
+            del vl_view_linestring
+
+        print('-I-> test_01_{0} [{1}]'.format(test_name,'end of test'))
+
+###############################################################################
+# Test Spatialite SpatialTables with XYZM values
+# - created with gdal autotest/auto/ogr_sql_sqlite.py spatialite_5()
+# contains 47 SpatialTables with (Multi-) Points/Linestrings/Polygons/GeometryCollection as XY,XYZ,XYM,XYZM
+# Test Geometries taken from /autotest/ogr/data/curves_line.csv and curves_polygon.csv
+# - to test the correct reading of: CircularString, CompoundCurve, CurvePolygon, MultiCurve and MultiSurface
+#
+
+    def test_02_OgrSpatialTableXYZM(self):
```

Comment:
As the documentation states, it is to check that all supported types can be read correctly.

In the case of **CircularString, CompoundCurve, CurvePolygon, MultiCurve and MultiSurface**, it was to check what promote logic was doing and when it was being called:

> When there are CurvePolygons, promote Polygons
> When there are CompoundCurves, promote LineStrings and CircularStrings

It turned out that it was never being called (at least from these sample files) and being rendered properly.

(It also looks as if I missed removing all gdal 1 portions)

So this test checks that the expected functionality works correctly.
After any changes (such as done on the 30.06.2016, https://hub.qgis.org/issues/15168), such checks would insure that **everything else** still works correctly.

To avoid this situation is very much a part ot the subject of this PR.